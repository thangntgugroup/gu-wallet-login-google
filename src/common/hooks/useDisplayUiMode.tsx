import React, { createContext, ReactNode, useContext } from "react";
import { DisplayUiModeType } from "../objects/object-types";

const DisplayUiModeContext = createContext<{
  displayUiMode: DisplayUiModeType;
}>({
  displayUiMode: DisplayUiModeType.Light,
});

interface DisplayUiModeProviderProps {
  children: ReactNode;
}

export const useDisplayUiMode = () => {
  const context = useContext(DisplayUiModeContext);
  return context.displayUiMode;
};

export const DisplayUiModeProvider = ({
  children,
}: DisplayUiModeProviderProps) => {
  const queryParams = new URLSearchParams(window.location.search);
  const displayUiMode = queryParams.get("uiMode") as DisplayUiModeType;

  return (
    <DisplayUiModeContext.Provider value={{ displayUiMode }}>
      {children}
    </DisplayUiModeContext.Provider>
  );
};
