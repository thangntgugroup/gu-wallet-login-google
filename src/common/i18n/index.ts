import i18next, { InitOptions } from 'i18next';
import { initReactI18next } from "react-i18next";

import commonEn from "./locales/en/common.json";
import commonJa from "./locales/ja/common.json";

const queryParams = new URLSearchParams(window.location.search);
const language = queryParams.get("language") || "en";

const i18nOptions: InitOptions = {
  resources: {
    en: {
      common: commonEn,
    },
    ja: {
      common: commonJa,
    },
  },
  defaultNS: 'common',
  lng: language,
  fallbackLng: "en",
  interpolation: {
    escapeValue: false,
  },
};

i18next.use(initReactI18next).init(i18nOptions);

export default i18next;
