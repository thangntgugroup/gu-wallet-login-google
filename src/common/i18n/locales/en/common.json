{
  "title": "Login with Web3Auth",
  "description": "Please log in with your GU account. If you don't have one, please create an account.",
  "sign_in": "Sign in G.U. Account with Web3Auth",
  "description_continue_google": "If you don't have a G.U. Account, it will be automatically created when you log in. In that case, it will be considered that you have agreed to the terms of use.",
  "continue_with_google": "Continue with Google",
  "cancel": "Cancel"
}
