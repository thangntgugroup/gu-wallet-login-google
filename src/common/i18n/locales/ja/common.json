{
  "title": "Web3Authでログイン",
  "description": "GUアカウントでログインしてください。お持ちでない方はアカウントを作成してください。",
  "sign_in": "Web3Authを使用してG.U. Accountにサインインする",
  "description_continue_google": "G.U.Accountをお持ちでない場合、ログインすると自動的にG.U.Accountが作成されます。その場合、利用規約に同意したものと見なします。",
  "continue_with_google": "Google を続ける",
  "cancel": "キャンセル"
}
