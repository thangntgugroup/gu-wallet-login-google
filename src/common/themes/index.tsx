import { createTheme, ThemeProvider } from '@material-ui/core';
import {
  ALICE_BLUE_2_COLOR,
  BLACK_COLOR,
  COMET_COLOR,
  ECLIPSE_COLOR,
  NIGHT_RIDER_2_COLOR,
  NOBEL_COLOR,
  WHITE_COLOR,
} from '../constants/colors';
import { useDisplayUiMode } from '../hooks/useDisplayUiMode';
import React from 'react';
import { DisplayUiModeType } from '../objects/object-types';

export const defaultTheme = createTheme({
  typography: {
    fontFamily: 'inherit',
  },
});

export const generateTheme = (mode: DisplayUiModeType) => {
  return mode === DisplayUiModeType.Dark
    ? createTheme({
        typography: {
          fontFamily: 'inherit',
        },
        palette: {
          type: 'dark',
          background: {
            default: NIGHT_RIDER_2_COLOR,
            paper: ECLIPSE_COLOR,
          },
          text: {
            primary: WHITE_COLOR,
            secondary: NOBEL_COLOR,
          },
        },
      })
    : createTheme({
        typography: {
          fontFamily: 'inherit',
        },
        palette: {
          type: 'light',
          background: {
            default: ALICE_BLUE_2_COLOR,
            paper: WHITE_COLOR,
          },
          text: {
            primary: BLACK_COLOR,
            secondary: COMET_COLOR,
          },
        },
      });
};

interface Props {}

export const CustomThemeProvider: React.FC<Props> = ({ children }) => {
  const displayUiMode = useDisplayUiMode();
  const theme = generateTheme(displayUiMode);

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
