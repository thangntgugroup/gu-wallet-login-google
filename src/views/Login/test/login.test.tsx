import * as React from "react";
import * as renderer from "react-test-renderer";
import View from "../index";
import { I18nextProvider } from "react-i18next";
import i18n from "../../../common/i18n";
import { DisplayUiModeProvider } from "../../../common/hooks/useDisplayUiMode";
import { CustomThemeProvider } from "../../../common/themes";

it("Login - renders correctly in English", () => {
  i18n.changeLanguage("en");
  const tree = renderer
    .create(
      <I18nextProvider i18n={i18n}>
        <DisplayUiModeProvider>
          <CustomThemeProvider>
            <View />
          </CustomThemeProvider>
        </DisplayUiModeProvider>
      </I18nextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("Login - renders correctly in Japanese", () => {
  i18n.changeLanguage("ja");
  const tree = renderer
    .create(
      <I18nextProvider i18n={i18n}>
        <DisplayUiModeProvider>
          <CustomThemeProvider>
            <View />
          </CustomThemeProvider>
        </DisplayUiModeProvider>
      </I18nextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
