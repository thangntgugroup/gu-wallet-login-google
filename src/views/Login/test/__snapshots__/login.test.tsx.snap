// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login - renders correctly in English 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-container-2"
  >
    <div
      className="makeStyles-title-3"
    >
      Login with Web3Auth
    </div>
    <div
      className="makeStyles-description-4"
    >
      Please log in with your GU account. If you don't have one, please create an account.
    </div>
    <div
      className="makeStyles-description-4"
    >
      Sign in G.U. Account with Web3Auth
    </div>
    <div
      className="makeStyles-buttonSignIn-5"
      onClick={[Function]}
    >
      <img
        alt=""
        className="makeStyles-buttonSignInIcon-6"
        src="google_login.png"
      />
      <p
        className="makeStyles-buttonSignInText-7"
      >
        Continue with Google
      </p>
    </div>
    <div
      className="makeStyles-description-4"
    >
      If you don't have a G.U. Account, it will be automatically created when you log in. In that case, it will be considered that you have agreed to the terms of use.
    </div>
    <div
      className="makeStyles-cancel-8"
      onClick={[Function]}
    >
      Cancel
    </div>
  </div>
</div>
`;

exports[`Login - renders correctly in Japanese 1`] = `
<div
  className="makeStyles-root-9"
>
  <div
    className="makeStyles-container-10"
  >
    <div
      className="makeStyles-title-11"
    >
      Web3Authでログイン
    </div>
    <div
      className="makeStyles-description-12"
    >
      GUアカウントでログインしてください。お持ちでない方はアカウントを作成してください。
    </div>
    <div
      className="makeStyles-description-12"
    >
      Web3Authを使用してG.U. Accountにサインインする
    </div>
    <div
      className="makeStyles-buttonSignIn-13"
      onClick={[Function]}
    >
      <img
        alt=""
        className="makeStyles-buttonSignInIcon-14"
        src="google_login.png"
      />
      <p
        className="makeStyles-buttonSignInText-15"
      >
        Google を続ける
      </p>
    </div>
    <div
      className="makeStyles-description-12"
    >
      G.U.Accountをお持ちでない場合、ログインすると自動的にG.U.Accountが作成されます。その場合、利用規約に同意したものと見なします。
    </div>
    <div
      className="makeStyles-cancel-16"
      onClick={[Function]}
    >
      キャンセル
    </div>
  </div>
</div>
`;
